src/wp-includes/class-wp-scripts.php

```diff
@@ -714,6 +781,262 @@ public function in_default_dir( $src ) {
 		return false;
 	}
 
+	/**
+	 * This overrides the add_data method from WP_Dependencies, to support normalizing of $args.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle Name of the item. Should be unique.
+	 * @param string $key    The data key.
+	 * @param mixed  $value  The data value.
+	 * @return bool True on success, false on failure.
+	 */
+	public function add_data( $handle, $key, $value ) {
+		if ( 'script_args' === $key ) {
+			$args = $this->get_normalized_script_args( $handle, $value );
+			if ( $args['in_footer'] ) {
+				parent::add_data( $handle, 'group', 1 );
+			}
+			return parent::add_data( $handle, $key, $args );
+		}
+		return parent::add_data( $handle, $key, $value );
+	}
+
+	/**
+	 * Checks all handles for any delayed inline scripts.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @return bool True if the inline script present, otherwise false.
+	 */
+	public function has_delayed_inline_script() {
+		foreach ( $this->registered as $handle => $script ) {
+			// Non standalone scripts in the after position, of type async or defer, are usually delayed.
+			$strategy = $this->get_intended_strategy( $handle );
+			if ( $this->is_non_blocking_strategy( $strategy ) &&
+				$this->has_non_standalone_inline_script( $handle, 'after' )
+			) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Normalize the data inside the $args parameter and support backward compatibility.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string        $handle Name of the script.
+	 * @param array         $args     {
+	 *      Optional. Additional script arguments. Default empty array.
+	 *
+	 *      @type boolean   $in_footer    Optional. Default false.
+	 *      @type string    $strategy     Optional. Values blocking|defer|async. Default 'blocking'.
+	 * }
+	 * @return array        Normalized $args array.
+	 */
+	private function get_normalized_script_args( $handle, $args = array() ) {
+		$default_args = array(
+			'in_footer' => false,
+			'strategy'  => 'blocking',
+		);
+
+		// Handle backward compatibility for $in_footer.
+		if ( true === $args ) {
+			$args = array( 'in_footer' => true );
+		}
```

Comment:
What if `$args` is `false`? Won't this result in an error below when `wp_parse_args()` is called? The `wp_parse_args()` is supposed to only accept a `string|array|object` for its `$args` param.

```suggestion
		if ( is_bool( $args ) ) {
			$args = array( 'in_footer' => $args );
		}
```

Reply:
More importantly, I would suggest we handle the backward compatibility implications of this entirely outside of the `WP_Scripts` class. The change of `$in_footer` to `$args` happened in the `wp_register_script()` and `wp_enqueue_script()` functions and does not really concern the interior of this class. I think the `get_normalized_script_args()` function here should, since it focuses on the _new_ "script args" group, only cater for the new _intended_ way of passing those, which is an array.

In other words, I'd suggest we change this method to only contain the defaults and the `wp_parse_args()` bit, and move the backward compatibility related boolean handling directly into `wp_register_script()` and `wp_enqueue_script()`. Technically that means the code is duplicated across two functions, but it's just 3 lines and since we're working with functions, we can't introduce a private utility function, and introducing a (by definition public) function just for that doesn't seem like a good idea.
```php
if ( ! is_array( $args ) ) {
	$args = array( 'in_footer' => (bool) $args );
}
```

Doing that decouples the backward compatibility handling from this class, which never had to deal with the `$in_footer` parameter before in the first place.

Reply:
I think this is addressed by https://github.com/10up/wordpress-develop/pull/64

Reply:
@felixarntz this is addressed in @westonruter 's PR https://github.com/10up/wordpress-develop/pull/64 and can be seen (for `wp_register_script`) [here](https://github.com/10up/wordpress-develop/pull/64/files#diff-bcaf11790153d7289e128d04ba774971e72864e51141e29f5f56701ab99a3a4aR181) and (for `wp_enqueue_script`) [here](https://github.com/10up/wordpress-develop/pull/64/files#diff-bcaf11790153d7289e128d04ba774971e72864e51141e29f5f56701ab99a3a4aR373)